### PR TITLE
API Lambda to generate pre-signed URLs for uploading files

### PR DESCRIPTION
**Pull request by briri**
New API function to support the generate of pre-signed URLs that COKI can use to send us match data files.

Initial part of #21.

Also fixed up some issues with the API function that pre-signs URLs to download DMP metadata files:
- Fixed name in package.json
- Removed unnecessary `sam_build_deploy.rb` script
- Fixed typo and remove unused imports
- Fixed broken tests

### CODE REVIEW

File: lambdas/api/get_dmps_downloads/src/__tests__/index.spec.ts [hunk 1]
```diff
@@ -124,17 +124,18 @@ describe('handler', () => {
...
     mockVerifyAPIGatewayLambdaAuthorizer.mockResolvedValue({ name: 'Tester' });
-    mockListObjects.mockResolvedValue([{ key: 'testOne' }, { key: 'testTwo' }]);
-    mockGetPresignedURL.mockResolvedValueOnce(expected[0]);
-    mockGetPresignedURL.mockResolvedValueOnce(expected[1]);
+    mockListObjects.mockResolvedValue([{ key: 'testOne.json' }, { key: 'testTwo.json' }]);
+    mockGetPresignedURL.mockResolvedValueOnce(expected['testOne.json']);
+    mockGetPresignedURL.mockResolvedValueOnce(expected['testTwo.json']);
     const response = await handler(mockEvent, mockContext, undefined);
 
+console.log(response.body)
```
Comment (jupiter007) [inline lambdas/api/get_dmps_downloads/src/__tests__/index.spec.ts:137]: L137: Did you intentionally leave `console.log()` here?

File: lambdas/api/template.yaml [hunk 2]
```diff
@@ -1030,6 +1010,66 @@ Resources:
       # Explicit retention time
       RetentionInDays: !Ref LogRetentionDays
 
+  # Get DMP metadata files
+  PutUploadsDmpsFunction:
+    Type: 'AWS::Serverless::Function'
+    Properties:
+      # CodeUri: 'put_dmps_uploads/dist/'
```
Comment (jupiter007) [inline lambdas/api/template.yaml:1017]: L1017: Do we still need this commented line?

File: lambdas/api/put_dmps_uploads/src/__tests__/index.spec.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
+const mockGetExport = jest.fn();
```
Comment (jupiter007) [inline lambdas/api/put_dmps_uploads/src/__tests__/index.spec.ts:1]: Is there a way to run these unit tests locally? Whenever I try I get the error "Cannot find module '@aws-sdk/client-cloudformation'".

File: lambdas/api/put_dmps_uploads/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
+import { lambdaRequestTracker } from 'pino-lambda';
```
Comment (jupiter007) [inline lambdas/api/put_dmps_uploads/src/index.ts:1]: Do we need to add the endpoint here? https://github.com/CDLUC3/dmsp_api_prototype/wiki/api-overview

**Review by jupiter007 — APPROVED** [review]
Just a couple of minor comments.

LGTM

File: lambdas/api/get_dmps_downloads/src/__tests__/index.spec.ts (hunk 1 above)
Comment (briri) [reply lambdas/api/get_dmps_downloads/src/__tests__/index.spec.ts:137]: no, will remove that

File: lambdas/api/template.yaml (hunk 2 above)
Comment (briri) [reply lambdas/api/template.yaml:1017]: Oh great catch. That commented out line is the correct one. I was trying to get it to compile from the Typescript source but it fails. I forgot to switch it back before pushing

File: lambdas/api/put_dmps_uploads/src/__tests__/index.spec.ts (hunk 3 above)
Comment (briri) [reply lambdas/api/put_dmps_uploads/src/__tests__/index.spec.ts:1]: yes, you need to be in the `lambdas/api/put_dmps_uploads` dir and run `npm install` first then `npm run test` that should install all of those dev dependencies.

File: lambdas/api/put_dmps_uploads/src/index.ts (hunk 4 above)
Comment (briri) [reply lambdas/api/put_dmps_uploads/src/index.ts:1]: 💯

File: lambdas/api/put_dmps_uploads/src/__tests__/index.spec.ts (hunk 3 above)
Comment (jupiter007) [reply lambdas/api/put_dmps_uploads/src/__tests__/index.spec.ts:1]: I did that, but still see that error for some reason.

Comment (briri) [reply lambdas/api/put_dmps_uploads/src/__tests__/index.spec.ts:1]: Hmm. try going into the `lambda/layers/dmptool/cloudformation` directory and run `npm install`. Then go back to the `lambda/api/put_dmps_uploads` dir and run the tests again. They reference those lambda layers directly when you run locally so I wonder if the error is actually coming from the layer

Comment (jupiter007) [reply lambdas/api/put_dmps_uploads/src/__tests__/index.spec.ts:1]: Thanks @briri. That worked

**Pull request merged**